scripts/commit.js

```diff
@@ -49,16 +68,94 @@ const getStagedCount = async () => {
 const MIN_CHAR_LENGTH = 2;
 const MAX_CHAR_LENGTH = 100;
 
+async function createCommit(answers) {
+  const refMessage = answers.ticket ? `ref: #${answers.ticket}\n\n` : '\n\n';
+  let commitMessage = `${answers.type}(${answers.scope}): ${answers.title}\n\n`;
+
+  if (answers.description) {
+    commitMessage += `\n\n${answers.description}\n\n`;
+  }
+
+  commitMessage += refMessage;
+
+  // Add breaking change footer if it exists
+  if (answers.breaking) {
+    commitMessage += `\n\nBREAKING CHANGE: ${answers.breaking}`;
+  }
```

Comment:
From @helabenkhalfallah  refactoring meeting I can suggest :) 

```suggestion

function createCommitMessage(answers) {
const commitMessage = [];

return {
withHeader() {}
withDescription() {}
withRefTicker() {}
withBreakingChange() {}
}

async function createCommit(answers) {
  const commitMessage = createCommitMessage(answer)
  .withHeader()
  .withDescription()
  .withRefTicker()
  .withBreakingChange()
  
  [...]
  }
```

Reply:
And the build step to return commitMessage.join('\n')